tldraw/tldraw
# 1666 embed scene

Describe what your pull request does. If you can, add GIFs or images showing the before and after of your change.

partially fixes #1666

https://github.com/user-attachments/assets/abc3b39a-d9cc-404f-90dc-01ee2bd7979e

This was done as part of side project saturday.

### Change type
 

### Test plan

1. Create a shape...
2. Export...
3. Embed scene
4. SVG
5. Delete everything in the scene
6. File...
7. Open file
8. Select the SVG file that you just saved
9. Notice that your scene is preserved

todos:
- [x] menu item
- [ ] doesn't actually seem to update the state? :-(
- [x] default translation
- [ ] do we want to call it "embed" or do we want another word?
- [ ] the rest of the translations
- [ ] do we want a new input group?
- [ ] should this be off by default
- [ ] also embed scene in png files
- [ ] some kind of UX audit of the whole file opening flow to make it obvious that you can only import 
- [ ] `git grep FIXME` and fix all the things

future extensions:
- [ ] hitting ctrl+s when you just opened a .tldr/.svg/.png with scene should save back into that file with that format (this was my main contribution to excalidraw in https://github.com/excalidraw/excalidraw/pull/3645)
- [ ] hitting ctrl+s when you've just exported with scene should re-save that file?

also:

### Release notes

- Can now embed the tldraw scene in svg and png files, and re-open them to edit
